src/config.cpp

```diff
@@ -546,11 +560,45 @@ class SimulationConfig::Parser
             auto& valueIt = it.value();
             const auto debugStr = fmt::format("report {}", it.key());
             parseMandatory(valueIt, "cells", debugStr, report.cells);
+            parseOptional(valueIt, "sections", report.sections);
             parseMandatory(valueIt, "type", debugStr, report.type);
+            parseOptional(valueIt, "scaling", report.scaling);
+            parseOptional(valueIt, "compartments", report.compartments);
+            parseMandatory(valueIt, "variable_name", debugStr, report.variableName);
```

Comment:
for `variable_name`, I see the spec says there is the possibility of comma separated values; it should probably be checked for correctness (ie: trailing `,`, double comma, etc)